src/sortStudents.ts

```diff
@@ -1,16 +1,56 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrades({ grades }: Student): number {
+  return grades.reduce((acc, grade) => acc + grade, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((student1: Student, student2: Student) => {
+    switch (sortBy) {
+      case SortType.Married:
+        return Number(student2.married) - Number(student1.married);
+
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? student1[sortBy].localeCompare(student2[sortBy])
+          : student2[sortBy].localeCompare(student1[sortBy]);
+
+      case SortType.Age:
+        return order === 'asc'
+          ? student1.age - student2.age
+          : student2.age - student1.age;
+
+      case SortType.AverageGrade:
+        return order === 'asc'
+          ? getAverageGrades(student1)
+            - getAverageGrades(student2)
+          : getAverageGrades(student2)
+            - getAverageGrades(student1);
```

Comment:
You are checking your sort type every time, but it is defined only once. So will be better if you will define the sort type firstly and return sorted values next

```suggestion
  const sortedStudens = [...students];
  
    switch (sortBy) {
      case SortType.Married:
        return sortedStudens.sort((student1: Student, student2: Student) => (
          Number(student2.married) - Number(student1.married)
        );

      case SortType.Name:
      case SortType.Surname:
 ....

      case SortType.Age:
        return order === 'asc'
....

      case SortType.AverageGrade:
        return order === 'asc'
...
```

Reply:
✔️